crates/rolldown_binding/src/options/plugin/binding_plugin_options.rs

```diff
@@ -84,12 +85,72 @@ pub struct BindingPluginOptions {
   pub write_bundle: Option<MaybeAsyncJsCallback<BindingOutputs, ()>>,
 }
 
-impl Debug for BindingPluginOptions {
+impl<const WEAK: bool> Debug for BindingPluginOptions<WEAK> {
   fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
     f.debug_struct("BindingPluginOptions").field("name", &self.name).finish_non_exhaustive()
   }
 }
 
+impl FromNapiValue for BindingPluginOptions<true> {
```

Comment:
Do we still need customization if we already make these option with `WEAK: true`?

Reply:
> Is there a way to make this work in napi_derive?

hmmm,  I think I can support the const gernic in `napi_derive`

Reply:
> Do we still need customization if we already make these option with `WEAK: true` like you above code?

Yes, because we need `BindingPluginOptions<true>` for main thread plugins and `BindingPluginOptions<false>` for plugins in worker threads.

If `pub struct BindingPluginOptions<const WEAK: bool = false>` is flipped to `pub struct BindingPluginOptions<const WEAK: bool = true>`, then we need `impl FromNapiValue for BindingPluginOptions<false>` and call `cb.ref(env)` instead.

Reply:
Nah. It's ok we mark main thread plugins `WEAK: true`.